doc/specs/#2046 - Command Palette.md

```diff
@@ -248,6 +290,77 @@ For example, consider the following list of commands:
 As the user types, we should **bold** each matching character in the command
 name, to show how their input correlates to the results on screen.
 
+### Commandline Mode
+
+Commandline mode is much simpler. In this mode, we'll simply display a text input,
+similar to the search box that's rendered for Action Mode. In this box, the
+user will be able to type a `wt.exe` style commandline. The user does not need
+to start this commandline with `wt` (or `wtd`, etc) - since we're already
+running in WT, the user shouldn't really need to repeat themselves.
+
+When the user hits <kbd>enter</kbd>, we'll attempt to parse the commandline. If
+we're successful in parsing the commandline, we can close the palette and
+dispatch the commandline. If the commandline had errors, we should reveal a text
+box with an error message below the text input. We'll leave the palette open
+with their entered command, so they can edit the commandline and try again. We
+should _probably_ leave the message up for a few seconds once they've begun
+editing the commandline, but eventually hide the message (ideally with a motion
+animation).
+
+### Switching Between Modes
+
+**TODO**: This is a topic for _discussion_.
+
+How do we differentiate Action Mode from Commandline Mode?
+
+I think there should be a character that the user types that switches the mode.
+This is reminiscent of how the command palette works in applications like VsCode
+and Sublime Text. The same UI is used for a number of functions. In the case of
+VsCode, when the user opens the palette, it's initially in a "navigate to file"
+mode. When the user types the prefix character `@`, the menu seamlessly switches
+to a "navigate to symbol mode". Similarly, users can use `:` for "go to line"
+and `>` enters an "editor command" mode.
+
+I believe we should use a similarly implemented UI. The UI would be in one of
+the two modes by default, and typing the prefix character would enter the other
+mode. If the user deletes the prefix character, then we'd switch back into the
+default mode.
+
+When the user is in Action Mode vs Commandline mode, if the input is empty
+(besides potentially the prefix character), we should probably have some sort of
+placeholder text visible to indicate which mode the user is in. Something like
+_"Enter a command name..."_ for action mode, or _"Type a wt commandline..."_ for
+commandline mode.
+
+Initially, I favored having the palette in Action Mode by default, and typing a
+`:` prefix to enter Commandline Mode. This is fairly similar to how tmux's
+internal command prompt works, which is bound to `<prefix>-:` by default.
+
+If we wanted to remain _similar_ to VsCode, we'd have no prefix character be the
+Commandline Mode, and `>` would enter the Action mode. I'd think that might

```

Comment:
I'm now doubly on board with `>` being the Commandline mode key, because that's very similar to what Powertoys Run picked for their "commandline mode" (whatever they're calling it)